File: PaymentApplication - using files/app/src/main/java/org/cso/android/app/payment/LoginActivity.java
package org.cso.android.app.payment;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import com.karandev.util.data.service.DataServiceException;

import org.cso.android.app.payment.data.service.PaymentApplicationDataService;
import org.cso.android.app.payment.data.service.dto.LoginInfoDTO;
import org.cso.android.app.payment.databinding.ActivityLoginBinding;
import static org.cso.android.app.payment.global.keys.BundleKeyKt.LOGIN_INFO;
import org.cso.android.app.payment.viewmodel.LoginActivityListenerViewModel;

import java.util.concurrent.ScheduledExecutorService;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class LoginActivity extends AppCompatActivity {
    private ActivityLoginBinding m_binding;

    @Inject
    PaymentApplicationDataService dataService;

    @Inject
    ScheduledExecutorService threadPool;


    private void saveLoginInfoUIThreadCallback(LoginInfoDTO loginInfo)
    {
        Toast.makeText(this, "Access granted", Toast.LENGTH_LONG).show();
        startActivity(new Intent(this, OperationsActivity.class).putExtra(LOGIN_INFO, loginInfo));

    }

    private void loginButtonClickedCallback()
    {
        try {
            var loginInfo = m_binding.getLoginInfo();

            if(dataService.checkAndSaveLoginInfo(loginInfo)){
                runOnUiThread(() -> { saveLoginInfoUIThreadCallback(loginInfo);});
            }else{
                Toast.makeText(this, "Access denied", Toast.LENGTH_LONG).show();
            }

        }
        catch (DataServiceException ex){
            runOnUiThread(() -> Toast.makeText(this, "Data Problem:" + ex.getMessage(), Toast.LENGTH_LONG).show());

        }
        catch (Throwable ignore){
            runOnUiThread(() -> Toast.makeText(this, "Problem Occured. Try again later", Toast.LENGTH_LONG).show());
        }
    }

    private void initBinding()
    {
        m_binding = DataBindingUtil.setContentView(this, R.layout.activity_login);
        m_binding.setViewModel(new LoginActivityListenerViewModel(this));
        m_binding.setLoginInfo(new LoginInfoDTO());
    }

    private void init()
    {
        initBinding();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        init();
    }

    public void loginButtonClicked()
    {
        threadPool.execute(() -> loginButtonClickedCallback()); // threadPool.execute(this::loginButtonClickedCallback);
    }

}

File: 006-MultipleActivity -1/app/src/main/java/org/cso/android/app/multipleactivity/LoginActivity.kt
package org.cso.android.app.multipleactivity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import org.cso.android.app.multipleactivity.databinding.ActivityLoginBinding
import org.cso.android.app.multipleactivity.keys.LOGIN_INFO
import org.cso.android.app.multipleactivity.viewmodel.LoginActivityListenersViewModel

import org.cso.android.app.multipleactivity.viewmodel.LoginInfo
import java.time.LocalDateTime


class LoginActivity : AppCompatActivity() {
    private lateinit var mBinding: ActivityLoginBinding

    /* alternative
    private fun startPaymentActivityCallBack(intent: Intent)
    {
        intent.putExtra(USER_NAME,mBinding.viewModel!!.username)
        startActivity(intent)

    }
    */

    private fun initViewModels()
    {
        mBinding.viewModel = LoginActivityListenersViewModel(this)
        mBinding.loginInfo = LoginInfo()
    }

    private fun initialize()
    {
        mBinding = DataBindingUtil.setContentView(this,R.layout.activity_login)
        initViewModels()

    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initialize()
    }

    fun loginButtonClicked()
    {
        Intent(this,PaymentActivity::class.java).apply {
            mBinding.loginInfo!!.loginDateTime = LocalDateTime.now()
            startActivity( putExtra(LOGIN_INFO,mBinding.loginInfo)) }
    }

}